tools/c7n_azure/c7n_azure/actions.py

```diff
@@ -52,28 +52,55 @@ def update_resource_tags(tag_action, resource, tags):
             if tag_action.manager.type == 'armresource':
                 raise NotImplementedError('Cannot tag generic ARM resources.')
 
+            # build a GenericResource object from the resource string returned by the Azure API
             az_resource = GenericResource.deserialize(resource)
```

Comment:
should just be able to use `resource.id` and not bother with the deserialize/az_resource at all perhaps.

Reply:
found this as well while trying to figure out the GenericResource properties to set. haven't committed / pushed yet though. will be included with next commit.